lib/gems/pending/util/mount/miq_swift_session.rb

```diff
@@ -0,0 +1,121 @@
+require 'util/mount/miq_generic_mount_session'
+
+class MiqSwiftSession < MiqGenericMountSession
+  def initialize(log_settings)
+    super(log_settings)
+    raise "username and password are required values!" if @settings[:username].nil? || @settings[:password].nil?
+    _scheme, _userinfo, @host, @port, _registry, @mount_path, _opaque, query, _fragment = URI.split(URI.encode(@settings[:uri]))
```

Comment:
codeclimate complaint:

> `URI.encode` method is obsolete and should not be used. Instead, use `CGI.escape`, `URI.encode_www_form` or `URI.encode_www_form_component` depending on your specific use case.

Reply:
@Fryguy I have seen this before myself, and it the suggestions don't actually work properly for what we are doing with it.

To avoid issues, and making sure we aren't introducing bugs, I have also just been keeping what is there for the time being.

Reply:
Agreed @NickLaMuro.  BTW @Fryguy this code duplicates what is done in all the other MountSession classes/subclasses as well.  If/When we change we should be consistent.